pkg/util/fxutil/logging/logging.go

```diff
@@ -7,55 +7,50 @@
 package logging
 
 import (
+	"os"
+
 	"go.uber.org/fx"
 	"go.uber.org/fx/fxevent"
 )
 
+// DefaultFxLoggingOption creates an fx.Option to configure the Fx logger, either to do nothing
+// (the default) or to log to the console (when TRACE_FX is set at `1`).
+func DefaultFxLoggingOption() fx.Option {
+	return fx.Options(
+		fx.WithLogger(
+			func() fxevent.Logger {
+				if os.Getenv("TRACE_FX") == "1" {
+					return &fxevent.ConsoleLogger{W: os.Stderr}
+				}
+				return fxevent.NopLogger
+			},
+		),
+	)
+}
+
 // Logger is a logger that logs fx events.
 type Logger interface {
 	Debug(v ...interface{})
 }
 
-type fxEventLogger struct {
-	logger func(v ...interface{})
-}
-
-// NewFxEventLoggerOption returns an fx option that provides a fxEventLogger.
-// Generic is used in order to not depends on the logger package.
-func NewFxEventLoggerOption[T Logger]() fx.Option {
-	// Note: The pointer in *T is needed for `optional:"true"`
-	return fx.Provide(func(logger *T) *fxEventLogger {
-		if logger == nil {
-			return nil
+// EnableFxLoggingOnDebug enables the logs for FX events when log_level is debug.
+// This function supposes that DefaultFxLoggingOption was used.
+// This function uses generic to avoid depending on the logger.
+// When TRACE_FX is set to 0, it will disable the logging.
+func EnableFxLoggingOnDebug[T Logger]() fx.Option {
+	return fx.Decorate(func(logger T) fxevent.Logger {
+		if os.Getenv("TRACE_FX") == "0" {
```

Comment:
Rather than using `if os.Getenv("TRACE_FX") == "0" {` which assumes that `TRACE_FX` env variable is set, which I'm not sure locally people has, what about using: 

```go 
if os.Getenv("TRACE_FX") == "1"` {
    return &fxevent.ConsoleLogger{W: fxEventLogger{logger: logger}}
} else {
    return fxevent.NopLogger
}

Reply:
The goal of this development is to log FX events when using `log_level: debug`. If for some reasons, the user don't want to see the FX events, they can set TRACE_FX at `0` to disabled it.
In your solution, logging FX events requires to have both `TRACE_FX:1` and `log_level: debug`

Reply:
I understand your point. 

For the explanation